packages/desktop/views/dashboard/collectibles/views/GalleryView.svelte

```diff
@@ -1,23 +1,63 @@
 <script lang="typescript">
-    import { Text, FontWeight, NftGalleryItem } from 'shared/components'
+    import { Text, TextInput, NftGalleryItem, TogglableButton, Icon, FontWeight } from 'shared/components'
     import { CollectiblesRoute, collectiblesRouter } from '@core/router'
-    import { INft, selectedAccountNfts } from '@core/nfts'
+    import { INft, nftSearchTerm, selectedAccountNfts, queriedNfts } from '@core/nfts'
     import { selectedNftId } from '../stores/selected-nft.store'
     import { localize } from '@core/i18n'
+    import { debounce } from '@core/utils'
 
+    let searchActive = false
+    let inputElement: HTMLInputElement
+
+    $: if (searchActive && inputElement) inputElement.focus()
+    $: searchValue = searchActive ? searchValue.toLowerCase() : ''
+    $: if ($selectedAccountNfts) {
+        debounce(() => {
+            $nftSearchTerm = searchValue
+        })()
+    }
     function handleNftClick(nft: INft): void {
         $selectedNftId = nft.id
         $collectiblesRouter.goTo(CollectiblesRoute.Details)
     }
 </script>
 
-<div class="flex flex-row text-left space-x-1 items-center">
-    <Text fontSize="text-14" fontWeight={FontWeight.semibold}>{localize('views.collectibles.gallery.title')}</Text>
-    <Text fontSize="text-14" fontWeight={FontWeight.semibold} color="gray-500">• {$selectedAccountNfts.length}</Text>
+<div class="flex flex-row justify-between">
+    <div class="flex flex-row text-left space-x-1 items-center">
+        <Text fontSize="text-14" fontWeight={FontWeight.semibold}>{localize('views.collectibles.gallery.title')}</Text>
+        <Text fontSize="text-14" fontWeight={FontWeight.semibold} color="gray-500">• {$selectedAccountNfts.length}</Text
```

Comment:
Do we want to show `$selectedAccountNfts.length` or do we want to show the number of returned NFTs from the search, so `$queriedNfts.length`?